docs/internal/proposals/error-handling/error-handling.md

```diff
@@ -0,0 +1,48 @@
+# Error handling in Director graphql
+
+In compass director, we have to somehow filter errors which should be visible to user.
+Basic list of errors which should be displayed to the user:
+* Internal error
+* Invalid Data
+* Not Unique
+* Not Found
+* Tenant Not Found
+* Insufficient Scopes
+
+This list can be split into 2 groups due to different handling:
+* internal errors
+* the rest of errors, which will be presented to the user
+
+In group of internal errors, are errors from:
+* external libraries
+* panics
+* postgreSQL, except errors like NotFound, NotUnique
+
+To deal with the errors we introduced custom errors, which contains error codes.
+
+## Proposed custom errors and theirs error codes (the error codes can change):
+
+| Error type           | Error code  |                            Description                                                      |
+|----------------------|-------------|---------------------------------------------------------------------------------------------|
+| Internal Error       | 1           | Error which cannot be handler in director                                                   |
+| Not Found            | 2           | Error indicate that given resource cannot be found and further processing is impossible     |
+| Not Unique           | 3           | Error indicates that given resource is not unique                                           |
+| Tenant Not Found     | 4           | Internal Tenant not found in director                                                       |
+| InvalidData          | 5           | The input data is invalid, error description should be delivered in error message           | 
+| ConstraintViolation  | 6           | Error which indicate that this operation can't happen because referenced resource not exist |
+| Insufficient Scopes  | 7           | Error which indicate insufficient scopes                                                    |
+
+NotFoundError can be triggered in mutation like `addPackage` to not existing application.
+
+ConstraintViolation can be replaced with `NotFound`  error.
```

Comment:
So should it be replaced? I do not like having two different ways of handling the same error, it would be confusing.